ansible/roles/vm_set/tasks/ptf_portchannel.yml

```diff
@@ -0,0 +1,30 @@
+---
+- name: Include variables for PTF containers
+  include_vars:
+    dir: "{{ playbook_dir }}/group_vars/ptf_host/"
+
+- name: Set ptf host
+  set_fact:
+    ptf_host: "ptf_{{ vm_set_name }}"
+    ptf_host_ip: "{{ ptf_ip.split('/')[0] }}"
+
+- name: Add ptf host
+  add_host:
+    hostname: "{{ ptf_host }}"
+    ansible_user: "{{ ptf_host_user }}"
+    ansible_ssh_host: "{{ ptf_host_ip }}"
+    ansible_ssh_pass: "{{ ptf_host_pass }}"
+    ansible_python_interpreter: "/usr/bin/python"
+    groups:
+      - ptf_host
+
+- name: find downlink portchannel configuration
+  set_fact:
+    portchannel_config: "{{ topology['DUT']['portchannel_config'] | default({})}}"
+
+- name: Start PTF portchannel
+  ptf_portchannel:
```

Comment:
>ptf_portchannel [](start = 2, length = 15)

What is the benefit of a new module vs using playbook directly?
Ansible has template and supervisor modules, I guess playbook will be easier.

Reply:
I'm not sure, I just follow the style of `exabgp`. I think we want to do similar task.